barretenberg/cpp/src/barretenberg/vm2/constraining/relations/bc_hashing.test.cpp

```diff
@@ -0,0 +1,231 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "barretenberg/vm2/constraining/testing/check_relation.hpp"
+#include "barretenberg/vm2/generated/columns.hpp"
+#include "barretenberg/vm2/generated/flavor_settings.hpp"
+#include "barretenberg/vm2/generated/relations/bc_hashing.hpp"
+#include "barretenberg/vm2/generated/relations/lookups_bc_hashing.hpp"
+#include "barretenberg/vm2/generated/relations/poseidon2_hash.hpp"
+#include "barretenberg/vm2/simulation/lib/contract_crypto.hpp"
+#include "barretenberg/vm2/simulation/poseidon2.hpp"
+#include "barretenberg/vm2/testing/macros.hpp"
+#include "barretenberg/vm2/tracegen/bytecode_trace.hpp"
+#include "barretenberg/vm2/tracegen/lib/lookup_builder.hpp"
+#include "barretenberg/vm2/tracegen/poseidon2_trace.hpp"
+#include "barretenberg/vm2/tracegen/precomputed_trace.hpp"
+#include "barretenberg/vm2/tracegen/test_trace_container.hpp"
+
+namespace bb::avm2::constraining {
+namespace {
+
+using tracegen::BytecodeTraceBuilder;
+using tracegen::Poseidon2TraceBuilder;
+using tracegen::TestTraceContainer;
+using FF = AvmFlavorSettings::FF;
+using C = Column;
+using bc_hashing = bb::avm2::bc_hashing<FF>;
+using poseidon2 = bb::avm2::poseidon2_hash<FF>;
+
+using bc_hashing_lookup_relation = bb::avm2::lookup_pos2_bc_hashing_relation<FF>;
+using bc_decomp_lookup_relation = bb::avm2::lookup_bc_decomp_relation<FF>;
+using length_iv_relation = bb::avm2::lookup_len_is_iv_relation<FF>;
+
+std::vector<FF> random_fields(size_t n)
+{
+    std::vector<FF> fields;
+    fields.reserve(n);
+    for (size_t i = 0; i < n; ++i) {
+        // FF::random_element() is really expensive, so we use std::rand
+        fields.push_back(FF(std::rand()));
+    }
+    return fields;
+}
+
+std::vector<uint8_t> random_bytes(size_t n)
+{
+    std::vector<uint8_t> bytes;
+    bytes.reserve(n);
+    for (size_t i = 0; i < n; ++i) {
+        bytes.push_back(static_cast<uint8_t>(rand() % 256));
+    }
+    return bytes;
+}
```

Comment:
Feel free to move to vm2/common/testing/fixtures.hpp.

I think I already do sth like this in bc_decomposition? (then deduplicate please).

Also I think it's ok if you use FF::random_element() for smallish sizes! In my case I needed it in a hot loop.